dvc/commands/run.py

```diff
@@ -70,19 +70,17 @@ def add_parser(subparsers, parent_parser):
         "--file", metavar="<filename>", help=argparse.SUPPRESS
     )
     run_parser.add_argument(
-        "--no-exec",
+        "--single-stage",
         action="store_true",
         default=False,
-        help="Only create dvc.yaml without actually running it.",
+        help=argparse.SUPPRESS,
     )
+    _add_common_args(run_parser)
```

Comment:
BTW shouldn't `-n|--name` be in `_add_common_args()`? @karajan1001

Reply:
I think we can

> BTW shouldn't `-n|--name` be added in `_add_common_args()`? It's a common arg both in `stage add` and `run`. Cc @karajan1001

I think we can do that.

> Also (💅🏼) should add_arguments() in repro/exp run use a similar private method name instead? _add_common_args() as well, even

I'm not quite understand here why this method to be a private method as it is used outside its file's scope.

Reply:
> BTW shouldn't `-n|--name` be added in `_add_common_args()`? It's a common arg both in `stage add` and `run`. Cc @karajan1001

Resolved.

Reply:
Never mind, rolling this change back. It breaks a bunch of tests. I think they were intentionally left separate to support some behavior that was deprecated but not actually removed (like `dvc run --single-stage`). Let's stick with the UI changes.